migration/transitioning.rst

```diff
@@ -0,0 +1,98 @@
+
+Migration tactics differ between the two main Citus use cases of multi-tenant applications and real-time analytics. The former requires fewer data model changes so we'll begin there.
+
+Multi-tenant Data Model
+=======================
+
+Citus is well suited to hosting B2B multi-tenant application data. In this model application tenants share a Citus cluster and a schema. Each tenant's table data is stored in a shard determined by a configurable tenant id column. Citus pushes queries down to run directly on the relevant tenant shard in the cluster, spreading out the computation. Once queries are routed this way they can be executed without concern for the rest of the cluster. These queries can use the full features of SQL, including joins and transactions, without running into the inherent limitations of a distributed system.
+
+This section will explore how to model for the multi-tenant scenario, including necessary adjustments to the schema and queries.
+
+The general process is
+
+1. Recreate the schema of the original database on the Citus coordinator node.
+2. Modify the schema for multi-tenant distribution.
+3. Copy the data from the original database to Citus.
+
+Schema Migration
+----------------
+
+Transitioning from a standalone database instance to a sharded multi-tenant system requires identifying and modifying three types of tables which we may term *per-tenant*, *reference*, and *global*. The distinction hinges on whether the tables have (or reference) a column serving as tenant id. The concept of tenant id depends on the application and who exactly are considered its tenants.
+
+Consider an example multi-tenant application similar to Etsy or Shopify where each tenant is a store. Here's a portion of a simplified schema:
+
+.. figure:: ../images/erd/mt-before.png
+   :alt: Schema before migration
+
+   (Underlined items are primary keys, italicized items are foreign keys.)
+
+In our example each store is a natural tenant. This is because storefronts benefit from dedicated processing power for their customer data, and stores do not need to access each other's sales or inventory. The tenant id is in this case the store id. We want to distribute data in the cluster in such a way that rows from the above tables in our schema reside on the same node whenever the rows share a store id.
+
+The first step is preparing the tables for distribution. Citus requires that primary keys contain the distribution column, so we must modify the primary keys of these tables and make them compound including a store id. Making primary keys compound will require modifying the corresponding foreign keys as well.
+
+In our example the stores and products tables are already in perfect shape. The orders table needs slight modification: updating the primary and foreign keys to include store_id. The line_items table needs the biggest change. Being normalized, it lacks a store id. We must add that column, and include it in the primary key constraint.
```

Comment:
The last discussed optional item here was to add the explicit `ALTER TABLE` statements to add the store_id and modify the PK and FK's. This will make the before/after pics more readable and understandable.